src/controllers/controllermanager.cpp

```diff
@@ -174,7 +178,20 @@ void ControllerManager::updateControllerList() {
 
     locker.relock();
     if (newDeviceList != m_controllers) {
+        // Check for keyboard controller
+        KeyboardController* pKeyboardController;
+        foreach (Controller* pController, newDeviceList) {
+            pKeyboardController = qobject_cast<KeyboardController*>(pController);
+            if (pKeyboardController) {
+                m_keyboardController = pKeyboardController;
+                connect(m_keyboardController, SIGNAL(presetLoaded(ControllerPresetPointer)),
+                        this, SLOT(slotKeyboardPresetChanged(ControllerPresetPointer)));
+            }
+        }
         m_controllers = newDeviceList;
+
+        // TODO(Tomasito) This unlock is redundant. locker, being a QMutexLocker
+        // ...            unlocks itself when destroyed
```

Comment:
It is a good practice to unlock before emit() this avoids deadlock if the connected slots are tying to lock again.